code/datums/outputs.dm

```diff
@@ -25,10 +25,17 @@
 		sound_output.volume = vol
 
 		if(vary)
-			if(frequency)
-				sound_output.frequency = frequency
-			else
-				sound_output.frequency = get_rand_frequency()
+			var/low = playback_rate - vary
+			var/high = playback_rate + vary
+			var/random_offset = (high - low) * rand()
+			playback_rate = low + random_offset
+
+		var/area/sound_area = get_area(turf_source)
+		if(sound_area && sound_area.sound_environment)
+			sound_output.environment = sound_area.sound_environment
+
+		if(receiver.override_sound_environment)
+			sound_output.environment = receiver.override_sound_environment
```

Comment:
You still have `environment` here.

Again, I've described the problem with how this functions.

Reply:
I actually just forgot to ever remove it.

Reply:
Great. Thank you!